config.c

```diff
@@ -35,16 +36,22 @@ write_default_config(FILE *f)
 }
 
 Config
-load_config(const char *fileName)
+load_config(const char *filename)
 {
-    FILE *f = fopen(fileName, "r");
-    if (f == NULL) {
-        f = fopen(fileName, "wr");
+    // NOTE: reading a file can fail for many reasons. 
+    // we should not be trying to create a new one in most of em
+	 if (access(filename, F_OK | R_OK)) {
+        printf("Config file %s does not exist, creating default\n", filename);
+        FILE *f = fopen(filename, "wr");
```

Comment:
if youre only writing a default config and re-opening it in readonly mode later anyways, why not just open as 'w'?